OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -290,25 +290,30 @@ public async Task<List<ShortEntityDto>> GetWorkshopListByEmployeeId(string emplo
     }
 
     /// <inheritdoc/>
-    public async Task<SearchResult<WorkshopProviderViewCard>> GetByProviderId(Guid id, ExcludeIdFilter filter)
+    public async Task<SearchResult<WorkshopProviderViewCard>> GetByProviderId(Guid id, WorkshopFilterTitle filter)
     {
         logger.LogInformation($"Getting Workshop by organization started. Looking ProviderId = {id}.");
 
-        filter ??= new ExcludeIdFilter();
-        ValidateExcludedIdFilter(filter);
+        filter ??= new WorkshopFilterTitle();
+        ValidateWorkshopTitleFilter(filter);
 
         var workshopBaseCardsCount = await workshopRepository.Count(whereExpression: x =>
-            filter.ExcludedId == null
-                ? (x.ProviderId == id)
-                : (x.ProviderId == id && x.Id != filter.ExcludedId)).ConfigureAwait(false);
+                (x.ProviderId == id) &&
+                (filter.ExcludedId == null || x.Id != filter.ExcludedId) &&
+                (string.IsNullOrEmpty(filter.SearchText) || x.Title.Contains(filter.SearchText)))
+                .ConfigureAwait(false);
 
         var workshops = await workshopRepository.Get(
                 skip: filter.From,
                 take: filter.Size,
-                includeExpression: includeFunc,
-                whereExpression: x => filter.ExcludedId == null
-                    ? (x.ProviderId == id)
-                    : (x.ProviderId == id && x.Id != filter.ExcludedId)).ToListAsync().ConfigureAwait(false);
+                includeProperties: includingPropertiesForMappingDtoModel,
```

Comment:
the previous version with `includeExpression: includeFunc` should be used now instead of include string (don't know why it did not suggest conflict over changing this property)